logstash_async/handler.py

```diff
@@ -104,6 +95,82 @@ def _setup_transport(self, **kwargs):
                 'Invalid transport path: must be an importable module path, '
                 'a class or factory function or an instance.')
 
+
+    # ----------------------------------------------------------------------
+    def _format_record(self, record):
+        self._create_formatter_if_necessary()
+        formatted = self.formatter.format(record)
+        if isinstance(formatted, str):
+            formatted = formatted.encode(self._encoding)  # pylint: disable=redefined-variable-type
+        return formatted + b'\n'
+
+    # ----------------------------------------------------------------------
+    def _create_formatter_if_necessary(self):
+        if self.formatter is None:
+            self.formatter = LogstashFormatter()
+
+    # ----------------------------------------------------------------------
+    def close(self):
+        self.acquire()
+        try:
+            self.shutdown()
+        finally:
+            self.release()
+        super().close()
+
+    # ----------------------------------------------------------------------
+    def shutdown(self):
+        self._close_transport()
+
+    # ----------------------------------------------------------------------
+    def _close_transport(self):
+        try:
+            if self._transport is not None:
+                self._transport.close()
+        except Exception as exc:
+            safe_log_via_print('error', u'Error on closing transport: {}'.format(exc))
+
+
+class AsynchronousLogstashHandler(SynchronousLogstashHandler):
+    """Python logging handler for Logstash. Sends events over TCP by default.
+    :param host: The host of the logstash server, required.
```

Comment:
Ideally we do not list here all arguments again but only those which are not handled by `SynchronousLogstashHandler`.

Reply:
I am not sure here how to handle this one here best, as I am not sure how to do that propber so e.g. IDEs understand that the other fields are also required for example, yet come from an inherited class

Reply:
I would assume IDEs reading the `:param` doc comments are able to handle inheritance.
Just remove the arguments which are inherited and maybe add a note above (just for humans) that there are more arguments from the base class.

Reply:
Done. As long as it gets merged I do everything :D